src/core/cdr/src/dds_cdrstream.c

```diff
@@ -1769,10 +1769,14 @@ static const uint32_t *dds_stream_skip_adr_default (uint32_t insn, char * __rest
   /* FIXME: currently only implicit default values are used, this code should be
      using default values that are specified in the type definition */
 
-  if (op_type_external (insn))
+  /* Free memory in sample and set pointer to null in case of optional or external member.
+     test for optional (which also gets the external flag) is added because string type
+     is the exception for this rule, that does not get the external flag */
+  if (op_type_external (insn) || op_type_optional (insn))
   {
-    *(void **) addr = NULL;
-    return dds_stream_skip_adr (insn, ops);
+    ops = stream_free_sample_adr(insn, data, ops);
+    *(char **) addr = NULL;
```

Comment:
For "external", this assignment is unnecessary because `stream_free_sample_adr` already does it. However, it does not store a null pointer after freeing the memory and this makes me wonder: if it is needed in the one case, is it then not needed in the other case?